src/lava/proc/plateau/models.py

```diff
@@ -0,0 +1,130 @@
+    vth_dend: int = LavaPyType(int, np.int32, precision=17)
+    vth_soma: int = LavaPyType(int, np.int32, precision=17)
+    up_dur: int = LavaPyType(int, np.uint16, precision=8)
+    up_state: int = LavaPyType(np.ndarray, np.uint16, precision=8)
+
+    def __init__(self, proc_params):
+        super(PyPlateauModelFixed, self).__init__(proc_params)
+        self.uv_bitwidth = 24
+        self.max_uv_val = 2 ** (self.uv_bitwidth - 1)
+        self.decay_shift = 12
+        self.decay_unity = 2 ** self.decay_shift
+        self.vth_shift = 6
+        self.act_shift = 6
+        self.isthrscaled = False
+
+    def scale_threshold(self):
+        self.effective_vth_dend = np.left_shift(self.vth_dend, self.vth_shift)
+        self.effective_vth_soma = np.left_shift(self.vth_soma, self.vth_shift)
+        self.isthrscaled = True
+
+    def subthr_dynamics(
+        self,
+        activation_dend_in: np.ndarray,
+        activation_soma_in: np.ndarray
+    ):
+        """Run the sub-threshold dynamics for both the dendrite and soma of the
+        neuron. Both use 'leaky integration'.
+        """
+        for v, dv, a_in in [
+            (self.v_dend, self.dv_dend, activation_dend_in),
+            (self.v_soma, self.dv_soma, activation_soma_in),
+        ]:
+            decayed_volt = np.int64(v) * (self.decay_unity - dv)
+            decayed_volt = np.sign(decayed_volt) * np.right_shift(
+                np.abs(decayed_volt), 12
+            )
+            decayed_volt = np.int32(decayed_volt)
+            updated_volt = decayed_volt + np.left_shift(a_in, self.act_shift)
+
+            neg_voltage_limit = -np.int32(self.max_uv_val) + 1
+            pos_voltage_limit = np.int32(self.max_uv_val) - 1
+
+            v[:] = np.clip(
+                updated_volt, neg_voltage_limit, pos_voltage_limit
+            )
+
+    def update_up_state(self):
+        """Decrements the up state (if necessary) and checks v_dend to see if
+        up state needs to be (re)set. If up state is (re)set, then v_dend is
+        reset to 0.
+        """
+        self.up_state[self.up_state > 0] -= 1
+        self.up_state[self.v_dend > self.effective_vth_dend] = self.up_dur
+        self.v_dend[self.v_dend > self.effective_vth_dend] = 0
+
+    def soma_spike_and_reset(self):
+        """Check the spiking conditions for the plateau soma. Checks if:
+            v_soma > v_th_soma
+            up_state > 0
+
+        For any neurons n that satisfy both conditions, sets:
+            s_out_buff[n] = True
+            v_soma = 0
+        """
+        s_out_buff = np.logical_and(
+            self.v_soma > self.effective_vth_soma,
+            self.up_state > 0
+        )
+        self.v_soma[s_out_buff] = 0
+
+        return s_out_buff
+
+    def run_spk(self):
+        """The run function that performs the actual computation during
+        execution orchestrated bgy a PyLoihiProcessModel using the
+        LoihiProtocol.
+        """
```

Comment:
typo in `orchestrated bgy a PyLoihiProcessModel`, should be `orchestrated by a PyLoihiProcessModel`